chain/client/src/debug.rs

```diff
@@ -382,87 +382,127 @@ impl ClientActor {
-                    block_height: height,
-                    block_producer,
-                    chunks: vec![],
-                    processing_time_ms: None,
-                    timestamp_delta: 0,
-                    gas_price_ratio: 1.0,
-                });
-            }
-
-            let block_producer = self
-                .client
-                .runtime_adapter
-                .get_block_producer(block.header().epoch_id(), block.header().height())
-                .ok();
-
-            let chunks = block
-                .chunks()
-                .iter()
-                .map(|chunk| DebugChunkStatus {
-                    shard_id: chunk.shard_id(),
-                    chunk_hash: chunk.chunk_hash(),
-                    chunk_producer: self
-                        .client
-                        .runtime_adapter
-                        .get_chunk_producer(
-                            block.header().epoch_id(),
-                            block.header().height(),
-                            chunk.shard_id(),
-                        )
-                        .ok(),
-                    gas_used: chunk.gas_used(),
-                    processing_time_ms: CryptoHashTimer::get_timer_value(chunk.chunk_hash().0)
-                        .map(|s| s.as_millis() as u64),
-                })
-                .collect();
 
-            blocks_debug.push(DebugBlockStatus {
-                block_hash: last_block_hash,
-                block_height: block.header().height(),
-                block_producer: block_producer,
-                chunks,
-                processing_time_ms: CryptoHashTimer::get_timer_value(last_block_hash)
-                    .map(|s| s.as_millis() as u64),
-                timestamp_delta: if last_block_timestamp > 0 {
-                    last_block_timestamp.saturating_sub(block.header().raw_timestamp())
-                } else {
-                    0
-                },
-                gas_price_ratio: block.header().gas_price() as f64 / initial_gas_price as f64,
-            });
-            last_block_hash = block.header().prev_hash().clone();
-            last_block_timestamp = block.header().raw_timestamp();
-            last_block_height = block.header().height();
+                let chunks = match &block {
+                    Some(block) => block
+                        .chunks()
+                        .iter()
+                        .map(|chunk| DebugChunkStatus {
+                            shard_id: chunk.shard_id(),
+                            chunk_hash: chunk.chunk_hash(),
+                            chunk_producer: self
+                                .client
+                                .runtime_adapter
+                                .get_chunk_producer(
+                                    block_header.epoch_id(),
+                                    block_header.height(),
+                                    chunk.shard_id(),
```

Comment:
You don't have to do it in this PR. Just note here the chunk may be created for a different height. It would also be nice if we display the chunk height here if it wasn't created for the block's height.

Reply:
And how are we handling the case if the chunk is missing?

Reply:
Yeah, I noticed the other issue for this. I'll handle it in a separate PR since this one is already too large haha

For chunk being missing, I haven't seen that case - is that possible?

Reply:
By chunks missing, I meant that the the block doesn't include new chunks for the shard. Here if a chunk is missing, the block will include the chunk header from the previous block, so you can check if a chunk is missing by check if `chunk.height_included() == block_header.height()`. 

Previously, I think the debug page display missing chunk info by showing the chunk has as 1111111. I just want to make sure that the new debug page will show missing chunks as well. If you just show the hash here, it's quite hard to detect that manually by comparing the chunk hash with the previous block's chunk hash.

Reply:
Ah yeah, the frontend does that now instead of the backend. It'll compare the chunk hash with the parent block's chunk hash.

Reply:
<img width="1769" alt="Screen Shot 2022-10-26 at 5 20 27 PM" src="https://user-images.githubusercontent.com/34969888/198146772-6290f63b-2001-426a-8cd7-6cd7182291fd.png">
I think the missing chunk is not handled correctly now. We can fix that in a following PR.

Reply:
Hmm from that screenshot I can't actually see why it's not handled correctly. The slot is grayed if the hash matches the hash of the same chunk in the *previous* block. In this case, we need to see what 32728 has in order to know whether the behavior is correct.